book-content/chapters/01-setup-typescript.md

```diff
@@ -0,0 +1,143 @@
+Before diving into TypeScript, let's take a moment to talk about its foundation — JavaScript.
+
+JavaScript is the language that makes web pages interactive. Any modern website will utilize some amount of it. And the more complex the site, the more complex the JavaScript.
+
+But, unlike other coding languages, JavaScript was not built for building complex systems.
+
+If you were building JavaScript apps in the 2000's, you were often having a bad time. Your IDE (integrated development environment) was lacking basic features. Autocomplete. Inline errors. There was no way to know if you were passing the right arguments to the right function. As users began demanding more complex experiences online, this made working with JavaScript a nightmare.
+
+This was especially true for refactoring code. If you had to change a function signature, you had to manually find and update every place that function was called throughout your entire codebase. This could take hours, and with no guarantee that you'd fixed everything before you pushed to production.
+
+## TypeScript's Beginnings
+
+As limitations like these became more apparent, developers started looking for a better way to write JavaScript.
+
+Around 2010, Microsoft noticed that a lot of their teams were using a community project called Script# (ScriptSharp) to build their JavaScript apps. This library allowed developers to write code in C#, and then turn it to JavaScript. C# had excellent features for building large applications - so it made the experience of building these apps more pleasant. In fact, many teams had found this was the only way they could build complex applications in large teams.
+
+Anders Hejlsberg, the creator of C#, was tasked to investigate this phenomenon. He was astonished. People were so annoyed with JavaScript that they were willing to code in a completely different language in order to get the powerful IDE features they were used to.
```

Comment:
I'm not sure if this is entirely true based on my recollection of the TS documentary but I could certainly be wrong. If I recall Anders joined the effort at some early stage after seeing an internal presentation on what was TS at the time. I definitely remember that he said things along those lines there:
> He was astonished. People were so annoyed with JavaScript that they were willing to code in a completely different language in order to get the powerful IDE features they were used to.

So dunno, maybe that presentation was about ScriptSharp? I'd double-check this with the documentary just to be sure.